### PR TITLE
Split workers #13

### PR DESCRIPTION
**Pull request by tom-butler**
Move worker config into a different folder.

Instead of blue / green logic, users can now run the nodes terraform again to create additional groups of nodes.

### CODE REVIEW

File: workspaces/datakube-dev/terraform.tfvars [hunk 1]
```diff
@@ -9,6 +9,10 @@ admin_access_CIDRs = {
   "Everywhere" = "0.0.0.0/0"
 }
 
+users = [
+  "user/tbutler",
```
Comment (roarmstrong) [inline workspaces/datakube-dev/terraform.tfvars]:
```suggestion
""
```

File: create.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    exit 1
+fi
+
+export WORKSPACE=$1
+
+# build network and EKS masters
+pushd infra
+rm -rf .terraform
+terraform init -backend-config ../workspaces/$WORKSPACE/backend.cfg 
+terraform plan -input=false -var-file="../workspaces/$WORKSPACE/terraform.tfvars"
```
Comment (woodcockr) [inline create.sh]: Seems unnecessary - what does plan do for you that apply will not do again?

File: create.sh [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+# build network and EKS masters
+pushd infra
+rm -rf .terraform
+terraform init -backend-config ../workspaces/$WORKSPACE/backend.cfg 
+terraform plan -input=false -var-file="../workspaces/$WORKSPACE/terraform.tfvars" 
+terraform apply -auto-approve -input=false -var-file="../workspaces/$WORKSPACE/terraform.tfvars" 
+
+# Configure local kubernetes config
+terraform output kubeconfig > ~/.kube/config-eks
```
Comment (woodcockr) [inline create.sh]: As I understand it these two lines do almost exactly the same thing. It seems unnecessary and redundant to create the separate config-eks file which you can already get from terraform. I suggest keeping only the second one.
```suggestion

```

File: create.sh [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+aws eks --region $(terraform output region) update-kubeconfig --name $(terraform output cluster_name)
+
+# Set up aws-auth
+terraform output config_map_aws_auth > aws-auth.yaml
+terraform output database_credentials > db-creds.yaml
```
Comment (woodcockr) [inline create.sh]: Perhaps do this further down where it is used for clarity?
```suggestion
```

File: create.sh [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+pushd nodes
+rm -rf .terraform
+terraform init -backend-config ../workspaces/$WORKSPACE/backend.cfg 
+terraform workspace new "$WORKSPACE-blue" || terraform workspace select "$WORKSPACE-blue"
+terraform plan -input=false -var-file="../workspaces/$WORKSPACE/terraform.tfvars" 
+terraform apply -auto-approve -input=false -var-file="../workspaces/$WORKSPACE/terraform.tfvars" 
+
+popd
+pushd infra
+
```
Comment (woodcockr) [inline create.sh]:
```suggestion
terraform output database_credentials > db-creds.yaml

```

File: create.sh [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+kubectl apply -f aws-auth.yaml
+popd
+
+# build worker nodes
+pushd nodes
+rm -rf .terraform
+terraform init -backend-config ../workspaces/$WORKSPACE/backend.cfg 
+terraform workspace new "$WORKSPACE-blue" || terraform workspace select "$WORKSPACE-blue"
+terraform plan -input=false -var-file="../workspaces/$WORKSPACE/terraform.tfvars"
```
Comment (woodcockr) [inline create.sh]: Same pattern, why plan then approve?

File: nodes/provider.tf [hunk 7]
```diff
@@ -0,0 +1,3 @@
+provider "aws" {
+  max_retries = 10
```
Comment (woodcockr) [inline nodes/provider.tf]: Add region variable
```suggestion
  region      = "${var.region}"
  max_retries = 10
```

File: nodes/variables.tf [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+  default     = ""
+}
+
+# Data sources
+
+variable "cluster_name" {}
+
+variable "owner" {
+  default = "opendatacube.org"
+}
```
Comment (woodcockr) [inline nodes/variables.tf]: Add region variable
```suggestion
variable "region" {

}

}
```

File: workspaces/datakube-prod/backend.cfg [hunk 9]
```diff
@@ -0,0 +1,4 @@
+bucket="dea-devs-tfstate"
```
Comment (woodcockr) [inline workspaces/datakube-prod/backend.cfg]: Will only leave one comment on the names and we are in progress fixing that up in #7

File: create.sh (hunk 2 above)
Comment (tom-butler) [reply create.sh]: Intent is to provide some rudimentary linting before trying to build the infrastructure, but given the time taken for a deploy I'll remove it. (tflint is better anyway)

Comment (woodcockr) [reply create.sh]: Almost said that might be the case but didn't want to presume.
Agree better to just one shot it, it apply will produce the same error if there is one

File: create.sh (hunk 6 above)
Comment (woodcockr) [reply create.sh]: As above

File: workspaces/datakube-prod/backend.cfg (hunk 9 above)
Comment (tom-butler) [reply workspaces/datakube-prod/backend.cfg]: It's just copied from the prior config, should be fixed when making names generic.

**Pull request merged**